src/components/shared/primary-button/primary-button.jsx

```diff
@@ -21,12 +22,16 @@ export default function PrimaryButton({
   isLeftArrow = false,
   hasBorder = true,
   isBold = false,
+  dark = false,
 }) {
+  const { route } = useRouter();
+
   const className = clsx(
     styles.primaryButton,
     classNames,
     hasBorder && styles.border,
-    isBold && styles.bold
+    isBold && styles.bold,
+    dark && route === '/mail' && styles.joinEmailButtonActive
```

Comment:
Take out the second part of the condition --> `route === '/mail'. You don't need this part. If the `dark` prop is true, show the dark styling, if its false, show its default stying.

Also, rename the `joinEMailButtonActvie` classname to something like `dark`.